test/integration/job/job_test.go

```diff
@@ -136,15 +515,45 @@ func TestMetrics(t *testing.T) {
 				Active: int(*jobObj.Spec.Parallelism),
 				Ready:  pointer.Int32(0),
 			}, wFinalizers)
-			if err, _ := setJobPodsPhase(ctx, clientSet, jobObj, v1.PodSucceeded, int(*jobObj.Spec.Parallelism)); err != nil {
-				t.Fatalf("Failed setting phase %s on Job Pod: %v", v1.PodSucceeded, err)
+
+			// we can optionally fail a pod
+			if tc.failedPodUpdate != nil {
+				if err, _ := updateJobPodsStatus(ctx, clientSet, jobObj, tc.failedPodUpdate, 1); err != nil {
+					t.Fatalf("Failed setting phase %s on Job Pod: %v", v1.PodFailed, err)
+				}
+			}
+
+			if tc.wantJobFailure {
```

Comment:
This is already tested in other tests.

Reply:
The purpose of this field was to select the appropriate waiting method rather than to do a check. Anyway, the field is gone after refactoring to extract the checks to dedicated tests.